### DISCUSSION TITLE
The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.

### DISCUSSION THREAD

**Discussion by ramba07** [prometheus/prometheus#10192] [Q&A]
**What did you do?**
I'm trying to build an image from the git repo source code of v2.31.0 branch.

**What did you expect to see?**
The image has to be built successfully.

**What did you see instead? Under which circumstances?**
'golint' error while running the Makefile.

**Environment**
CentOS

* System information:
Linux 3.10.0-1127.18.2.el7.x86_64 x86_64

* Prometheus version:
Not installed.

* Alertmanager version:
Not installed.

* Prometheus configuration file:
```
https://github.com/prometheus/prometheus/blob/main/.circleci/config.yml
```

* Alertmanager configuration file:
```
NA
```

* Logs:
```
NA
```

**Comment by LeviHarrison** [comment]
`golint` was replaced for that reason in https://github.com/prometheus/prometheus/pull/9571, but the patch isn't in that release. Linting isn't required to build a Docker image, so you can just follow the [Docker instructions](https://github.com/prometheus/prometheus#building-the-docker-image), which shouldn't fail.
